File: backend/crates/akashic-extraction/tests/fixtures/kotlin/01-class.kt
class Counter(private val start: Int) {
    var count: Int = 0
    private val label: String = "counter"

    constructor() : this(0)

    fun increment() {
        count = count + 1
        bump()
    }

    private fun bump() {
        count = count + 1
    }

    internal fun reset() {
        count = start
    }
}

File: backend/crates/akashic-extraction/tests/fixtures/objc/01-class.m
@interface Counter : NSObject
- (void)increment;
- (void)setValue:(int)v andLabel:(NSString *)l;
@end

@implementation Counter
- (void)increment {
    self.count += 1;
}

- (void)setValue:(int)v andLabel:(NSString *)l {
    [self increment];
    self.value = v;
    self.label = l;
}
@end
